src/protocols/secure_channel/SimpleSessionResumptionStorage.cpp

```diff
@@ -0,0 +1,273 @@
+/*
+ *    Copyright (c) 2022 Project CHIP Authors
+ *    All rights reserved.
+ *
+ *    Licensed under the Apache License, Version 2.0 (the "License");
+ *    you may not use this file except in compliance with the License.
+ *    You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+
+/**
+ *    @file
+ *      This file defines the CHIP CASE Session object that provides
+ *      APIs for constructing a secure session using a certificate from the device's
+ *      operational credentials.
+ */
+
+#include <protocols/secure_channel/SimpleSessionResumptionStorage.h>
+
+#include <lib/support/Base64.h>
+#include <lib/support/SafeInt.h>
+
+namespace chip {
+
+constexpr TLV::Tag SimpleSessionResumptionStorage::kIndexContentTag;
+constexpr TLV::Tag SimpleSessionResumptionStorage::kFabricIndexTag;
+constexpr TLV::Tag SimpleSessionResumptionStorage::kPeerNodeIdTag;
+constexpr TLV::Tag SimpleSessionResumptionStorage::kResumptionIdTag;
+constexpr TLV::Tag SimpleSessionResumptionStorage::kSharedSecretTag;
+constexpr TLV::Tag SimpleSessionResumptionStorage::kCATTag;
+
+const char * SimpleSessionResumptionStorage::StorageKey(DefaultStorageKeyAllocator & keyAlloc, const ScopedNodeId & node)
+{
+    return keyAlloc.FabricSession(node.GetFabricIndex(), node.GetNodeId());
+}
+
+const char * SimpleSessionResumptionStorage::StorageKey(DefaultStorageKeyAllocator & keyAlloc, ConstResumptionIdView resumptionId)
+{
+    char resumptionIdBase64[BASE64_ENCODED_LEN(resumptionId.size()) + 1];
+    auto len                = Base64Encode(resumptionId.data(), resumptionId.size(), resumptionIdBase64);
+    resumptionIdBase64[len] = '\0';
+    return keyAlloc.SessionResumption(resumptionIdBase64);
+}
+
+CHIP_ERROR SimpleSessionResumptionStorage::SaveIndex(const SessionIndex & index)
+{
+    uint8_t buf[MaxIndexSize()];
+    TLV::TLVWriter writer;
+    writer.Init(buf);
+
+    TLV::TLVType arrayType;
+    ReturnErrorOnFailure(writer.StartContainer(kIndexContentTag, TLV::kTLVType_Array, arrayType));
```

Comment:
Why not anonymous tag here?  We know this is an index.... and this is producing invalid TLV by using a context tag on the outermost element.